src/services/repo-index/repo-index.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from "@nestjs/common";
+import { Connection, Repository } from "typeorm";
+
+import { Logger } from "../../core";
+import { RepoReferenceRecord } from "../../models";
+
+export interface RepoReference {
+  readonly path: string;
+  readonly lastUse: number;
+  readonly lastFetch?: number;
+}
+
+const FETCH_CACHE_EXPIRY = 30_000; // 30s;
+
+@Injectable()
+export class RepoIndexService {
```

Comment:
>RepoIndexService [](start = 13, length = 16)

comments on what it does

Reply:
same for all the code

---
In reply to: [300110784](https://github.com/Azure/git-rest-api/pull/51#discussion_r300110784) [](ancestors = 300110784)